python/cuda_cccl/cuda/cccl/parallel/experimental/_utils/temp_storage_buffer.py

```diff
@@ -0,0 +1,75 @@
+import functools
+from types import SimpleNamespace
+
+import numpy as np
+
+from cuda.bindings import driver
+from cuda.core.experimental import Device, DeviceMemoryResource
+from cuda.core.experimental._utils.cuda_utils import handle_return
+
+
+class TempStorageBuffer:
+    """
+    Simple wrapper type around the memory allocation used for temporary storage,
+    exposing __cuda_array_interface__ and some other attributes for fast access.
+    """
+
+    def __init__(self, size: int):
+        # TODO: just use DeviceMemoryResource once cuda.core is updated
+        # to increase the default mempool threshold by default
+        mr = _DeviceMemoryResourceWithIncreasedThreshold(Device().device_id)
+        self._buf = mr.allocate(size)
+        self._ptr = int(self._buf.handle)
+
+        # other cuda_array_interface attributes
+        self._shape = (size,)
+        self._strides = (1,)
+        self._dtype = np.uint8
+
+        # attributes for fast path access in protocols.py
+        self.nbytes = size
+        self.data = SimpleNamespace(
+            ptr=self._ptr,
+        )
+
+    def __cuda_array_interface__(self):
+        return {
+            "data": (self._ptr, False),
+            "shape": self._shape,
+            "strides": self._strides,
+            "typ": self._dtype,
```

Comment:
I think this should be `typestr`, and it should return a str:
```suggestion
            "typestr": "|u1",
```
ex: 
```python
>>> import numpy as np
>>> uint8 = np.dtype(np.uint8)
>>> uint8.str
'|u1'
```

Reply:
Fixed